### PR TITLE
mobile efolder fix

### PR DESCRIPTION
**Pull request by aherzberg**
## Summary
stubbed data was not correct causing errors in staging. This corrects that.

I've also adds document_ids to claims show endpoint so docs in the efolder can be linked to specific claims. 
## Related issue(s)
https://github.com/department-of-veterans-affairs/va-mobile-app/issues/9774
https://github.com/department-of-veterans-affairs/va-mobile-app/issues/10029

## Testing done

- [ ] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: modules/mobile/app/models/mobile/v0/adapters/efolder.rb
Comment (aherzberg) [inline modules/mobile/app/models/mobile/v0/adapters/efolder.rb:1]: stubbed efolder spec data was incorrect

File: modules/mobile/app/models/mobile/v0/claim_event_timeline.rb
Comment (aherzberg) [inline modules/mobile/app/models/mobile/v0/claim_event_timeline.rb:1]: to match old EVSS formatting, untracked documents are created as an event, not a claim document.

**Review by digitaldrk — APPROVED** [review]

File: modules/mobile/docs/schemas/claimEventTimeline.yml [hunk 1]
```diff
@@ -57,4 +57,9 @@ properties:
     example: null
   filename:
     type: string
-    example: '7B434B58-477C-4379-816F-05E6D3A10487.pdf'
\ No newline at end of file
+    example: '7B434B58-477C-4379-816F-05E6D3A10487.pdf'
+  document_id:
+    type: string
+    description: 'Only used for other_documents_list type event for untracked documents'
```
Comment (jperk51) [inline modules/mobile/docs/schemas/claimEventTimeline.yml:63]: I think we want this doc id included for all docs not just untracked ones

Comment (aherzberg) [reply modules/mobile/docs/schemas/claimEventTimeline.yml:63]: tracked documents are already being tracked within the `modules/mobile/docs/schemas/claimDocument.yml`. Tracked items has a `documents` array that i've added document_id to. Untracked items are generated as a generic event so I had to add the document_id to the `eventTimeline` so it'll be null most of the time when not an untracked document. its not pretty but this is the format FE expects

**Review by jperk51 — APPROVED** [review]
lgtm

**Review by rmtolmach — APPROVED** [review]

**Pull request merged**